### PR TITLE
Add section on key management.

### PR DESCRIPTION
**Pull request by msporny**
This PR attempts to address issue #1284 by adding a section on key management. /cc @kdenhartog

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/vc-data-model/pull/1323.html" title="Last updated on Oct 31, 2023, 1:56 PM UTC (c764728)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/vc-data-model/1323/cc5cd92...c764728.html" title="Last updated on Oct 31, 2023, 1:56 PM UTC (c764728)">Diff</a>

### CODE REVIEW

File: index.html
Comment (brianorwhatever) [inline index.html:5022]: does this say signing is an "other purpose"?

File: index.html
Comment (dlongley) [inline index.html:5021]: Perhaps this will address @brianorwhatever's concern below:

```suggestion
[[NIST-SP-800-57-Part-1]], a private signing key is not to be used for multiple
```

**Review by dlongley — APPROVED** [review]

File: index.html
Comment (brianorwhatever) [reply index.html:5021]: yes, this feels better grammatically

**Review by decentralgabe — APPROVED** [review]

File: index.html
Comment (TallTed) [inline index.html:5017]:
```suggestion
on the quality and protection of their <em>private signing keys</em>. Guidance in
```

File: index.html
Comment (TallTed) [inline index.html:5031]: The keys are not advised; their administrators/controllers are.
```suggestion
[[NIST-SP-800-57-Part-1]] strongly advises that private signing keys and
<em>public verification keys</em> have limited <em>cryptoperiods</em>, where
a <em>cryptoperiod</em> is &quot;the time span during which a specific key is
authorized for use by legitimate entities or the keys for a given system will
remain in effect.&quot; [[NIST-SP-800-57-Part-1]] gives extensive
guidance on cryptoperiods for different key types under different situations,
and generally recommends a 1-3 year cryptoperiod for a private signing key.
```

File: index.html
Comment (TallTed) [inline index.html:5038]:
```suggestion
revocation. Although this section focuses primarily on the security of the
private signing key, [[NIST-SP-800-57-Part-1]] also highly recommends
confirmation of the validity of all public keys before using them.
```

**Review by kdenhartog — APPROVED** [review]
++ from me. Using NIST for key management recommendations is a good way to handle this without reinventing the wheel

**Review by OR13 — CHANGES_REQUESTED** [review]
Our controller document (that we intend to pull our from data integrity), discusses key agreement: https://w3c.github.io/vc-data-integrity/#key-agreement

On the WICG call yesterday, it was mentioned that responses from wallets are encrypted.

Early on we discussed that presentations might be encrypted, and it seems like that is the direction the industry is headed based on the WICG call.

Please adjust language to also cover encryption of presentations.

Ideally we could also provide full examples of encrypted presentations if we think that encrypting them is a best practice, or we would comment that presentations must be be made over unsecured channels.

File: index.html
Comment (msporny) [inline index.html:5022]:
```suggestion
[[NIST-SP-800-57-Part-1]], a private signing key is not to be used for multiple
purposes, such as the same key used for both encryption and signing.
```

File: index.html
Comment (msporny) [reply index.html:5022]: I tried to clarify below... does that change answer the question you're asking, @brianorwhatever ?

**Comment by msporny** [comment]
@OR13 wrote:
> Please adjust language to also cover encryption of presentations.
> Ideally we could also provide full examples of encrypted presentations if we think that encrypting them is a best practice, or we would comment that presentations must be be made over unsecured channels.

The issue raised, nor this PR that addresses that issue, concerns encryption of presentations. Please raise a new issue if you'd like that issue covered, addressed. This PR is purely about key management (not re-using the same key for different purposes). We explicitly say in the PR that a key shouldn't be used for both signing and encryption at the same time. 

Requesting a re-review given that this PR does not concern the changes you are requesting.

File: index.html
Comment (brianorwhatever) [reply index.html:5022]: yes this is more clear now thank you.

File: index.html
Comment (TallTed) [reply index.html:5022]:
```suggestion
[[NIST-SP-800-57-Part-1]], a private signing key is not to be used for multiple
purposes, e.g., a private signing key is not to be used for encryption as well
as signing.
```

**Review by TallTed — CHANGES_REQUESTED** [review]
Editorial tweak for clarity

**Comment by kdenhartog** [comment]
> Our controller document (that we intend to pull our from data integrity), discusses key agreement: [w3c.github.io/vc-data-integrity/#key-agreement](https://w3c.github.io/vc-data-integrity/#key-agreement)
> 
> On the WICG call yesterday, it was mentioned that responses from wallets are encrypted.
> 
> Early on we discussed that presentations might be encrypted, and it seems like that is the direction the industry is headed based on the WICG call.
> 
> Please adjust language to also cover encryption of presentations.
> 
> Ideally we could also provide full examples of encrypted presentations if we think that encrypting them is a best practice, or we would comment that presentations must be be made over unsecured channels.

I'm generally in support of what's requested here, but think it's going to be easier to discuss that topic separately after this is merged. I definitely think this is a topic that should be merged pre-rec stage as well (e.g. don't defer it to next version/maintainence), but it's nuanced enough of a topic that it's worth establishing consensus on it separately to this one. Especially if it's not requiring normative changes.

**Comment by msporny** [comment]
@OR13 it is not clear if you are going to block this PR because it doesn't speak to encryption and presentations (which is not the purpose of this PR). PING, represented by @kdenhartog, has requested that we address #1284 with this PR and merge it and has suggested that we open a new issue to address your concern. Would that work for you?

**Review by TallTed — APPROVED** [review]

**Comment by brentzundel** [comment]
As discussed during the 31 October meeting, this PR is being merged. 
The requested changes have been made. 
It is the group's understanding that there is nothing blocking the changes being made in this PR.
Additional possible changes to the spec can be tracked in further issues and PRs.

**Comment by msporny** [comment]
Editorial, multiple reviews, changes requested and made, no objections, merged.

**Pull request merged**
